src/components/SettlementButton.js

```diff
@@ -88,43 +100,64 @@ const defaultProps = {
     },
 };
 
-class SettlementButton extends React.Component {
-    componentDidMount() {
+function SettlementButton({
+    addDebitCardRoute,
+    addBankAccountRoute,
+    anchorAlignment,
+    betas,
+    buttonSize,
+    chatReportID,
+    currency,
+    enablePaymentsRoute,
+    iouReport,
+    isDisabled,
+    isLoading,
+    formattedAmount,
+    nvp_lastPaymentMethod,
+    onPress,
+    policyID,
+    shouldShowPaymentOptions,
+    shouldShowPaypal,
+    style,
+}) {
+    const {translate} = useLocalize();
+    const {isOffline} = useNetwork();
+
+    useEffect(() => {
         PaymentMethods.openWalletPage();
-    }
+    }, []);
 
-    getButtonOptionsFromProps() {
+    const getButtonOptionsFromProps = () => {
```

Comment:
I'm thinking that this should be memoized, we only care about the end value, and there are a lot of stuff going on here. 
If we memoize there are a couple of things we need to tackle, `betas` and `nvp_lastPaymentMethod` aren't stable default values. I would update the default values with stable ones like
```
const EMPTY_ARRAY = Object.freeze([]);
const EMPTY_OBJECT = Object.freeze({});

 betas: EMPTY_ARRAY,
 nvp_lastPaymentMethod: EMPTY_OBJECT,
 ```
First I will check how many times this component re-renders, to check if worth the change.
Then, I would double-check what Onyx returns, maybe we can make it stable too